src/app/core/entity/schema-datatypes/datatype-month.ts

```diff
@@ -40,12 +40,14 @@ export const monthEntitySchemaDatatype: EntitySchemaDatatype = {
     );
   },
 
-  transformToObjectFormat: (value) => {
-    value = value
+  transformToObjectFormat: (value: string) => {
+    const values = value
       .toString()
       .replace(/-(\d)-/g, "-0$1-")
-      .replace(/-(\d)$/g, "-0$1");
-    const date = new Date(value);
+      .replace(/-(\d)$/g, "-0$1")
```

Comment:
why do we still need the regex for leading zeros here if we work with the different constructor anyway?